packages/contracts-bedrock/contracts/periphery/OpStackChainRegistry.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+/**
+ * @title  OpStackChainRegistry
+ * @notice OpStackChainRegistry acts as a way to register and query
+ *         contract addresses for an OP Stack chain
+ */
+contract OpStackChainRegistry {
+
+    error OnlyDeploymentAdmin();
+
+    /**
+     * @notice Struct representing a deployment
+     *         on an OP Stack chain
+     */
+    struct Deployment {
+        string deploymentName;
+        address deploymentAdmin;
```

Comment:
I think it'd be worth having a function that allows an admin to transfer ownership over a deployment to a new address.

Reply:
Would be cool to include a pre-commit/timelock to ownership transfers here (maybe now, maybe in the future). Most people will use proxy addresses for the deployment entries, so upgrades can still happen in the meantime via the `ProxyAdmin` owner multisig. Longer term, it'd be awesome to consider the `ProxyAdmin` owner as the `Deployment`'s `deploymentAdmin`.

Reply:
Yeah for now this is basically a social consensus hack so there aren't any security assumptions being made in this contract, other than the fact that off-chain infrastructure (like the SDK) may bootstrap itself from these addresses. Deployers probably want to be careful with the owner of this contract no matter what.

Reply:
> Would be cool to include a pre-commit/timelock to ownership transfers here (maybe now, maybe in the future). Most people will use proxy addresses for the deployment entries, so upgrades can still happen in the meantime via the `ProxyAdmin` owner multisig. Longer term, it'd be awesome to consider the `ProxyAdmin` owner as the `Deployment`'s `deploymentAdmin`.

What would you think of a `mapping(string => mapping(address => boo)) = registeredAdmins` to keep track of a list of admins per deployment instead? Not much overhead needed and would allow multiple admins/addresses to manage a given deployment.

Would only additionally require
```solidity
// Checks to see at least 2 admin exist (in case od removal
// so no deployment is left orphaned
modifier twoAdmins(string memory _deployment, address _admin);

// Maintains map of deploymentNames => adminAddress => admin Exists
mapping(string => mapping(address => boo)) = registeredAdmins;

// Maintains map of deploymentName => deploymentExists
mapping(string => bool) deploymentExists;

function addAdmin(string memory _deployment, address _admin);

function revokeAdmin(string memory _deployment, address _admin) twoAdmins(_deployment, _admin);
```